src/Broadway/EventStore/DBALEventStore.php

```diff
@@ -231,4 +235,80 @@ private function convertStorageValueToIdentifier($id)
 
         return $id;
     }
+
+    public function visitEvents(EventVisitorInterface $eventVisitor, Criteria $criteria = null)
+    {
+        $statement = $this->prepareVisitEventsStatement($criteria ?: Criteria::create());
+        $statement->execute();
+
+        while ($row = $statement->fetch()) {
+            if ($this->useBinary) {
+                $row['uuid'] = $this->convertStorageValueToIdentifier($row['uuid']);
+            }
+
+            $domainMessage = $this->deserializeEvent($row);
+
+            $eventVisitor->doWithEvent($domainMessage);
+        }
+    }
+
+    private function prepareVisitEventsStatement(Criteria $criteria)
+    {
+        list ($where, $bindValues, $bindValueTypes) = $this->prepareVisitEventsStatementWhereAndBindValues($criteria);
+        $query = 'SELECT uuid, playhead, metadata, payload, recorded_on
+            FROM ' . $this->tableName . '
+            ' . $where . '
+            ORDER BY id ASC';
```

Comment:
See the comment I just added on this. `prepareVisitEventsStatementWhereAndBindValues` requires access to `isBinary` and `convertIdentifierToStorageValue` which means either refactoring this class in a larger way, exposing those variables via public methods, or duplicating the code in `DBALCriteriaQuery`. I don't think I have the time to do any of these right now (though public methods would probably be pretty trivial) so if this is super important this is probably a good candidate for someone else taking over to submit as a PR against my PR before we merge it.